src/chat/model_selector_list.rs

```diff
@@ -134,7 +134,9 @@ impl Widget for ModelSelectorList {
         cx.begin_turtle(walk, self.layout);
 
         if self.visible {
-            self.draw_items(cx, &store.downloads.downloaded_files);
+            if let Some(loaded_file) = store.get_loaded_downloaded_file() {
```

Comment:
Here is an issue. It could happen that there is no `loaded_file`, for this reason is an `Option` value.

You can replicate this corner case if you go to "My Models" and remove the current active model, and go back to the chat screen:

<img width="1421" alt="image" src="https://github.com/user-attachments/assets/6272b894-b71e-4de7-bc09-4e26aae682bd">

Looks like you can send the result of `store.get_loaded_downloaded_file()` and attempt to unwrap it in the `draw_items` function.